server/src/main/java/org/elasticsearch/action/fieldcaps/TransportFieldCapabilitiesAction.java

```diff
@@ -325,9 +326,28 @@ private static FieldCapabilitiesResponse merge(
         } else {
             collectResponseMap(responseMapBuilder, responseMap);
         }
+
+        // the merge method is only called on the primary coordinator for cross-cluster field caps, so we
+        // log relevant "5xx" errors that occurred in this 2xx response to ensure they are only logged once
+        for (FieldCapabilitiesFailure value : failures) {
+            Exception e = value.getException();
+            if (shouldLogException(e)) {
+                SEARCH_LOGGER.warn("Field caps [TransportFieldCapabilitiesAction] Exception for indices " + value.getIndices(), e);
+            }
+        }
         return new FieldCapabilitiesResponse(indices, Collections.unmodifiableMap(responseMap), failures);
     }
 
+    private static boolean shouldLogException(Exception e) {
+        // ConnectTransportExceptions are thrown when a cluster marked with skip_unavailable=false are not available for searching
+        // (Clusters marked with skip_unavailable=false return a different error that is considered a 4xx error.)
+        // In such a case, the field-caps endpoint returns a 200 (unless all clusters failed).
+        // To keep the logs from being too noisy, we choose not to log the ConnectTransportException here.
+        return e instanceof org.elasticsearch.transport.ConnectTransportException == false
```

Comment:
is the behaviour around ConnectTransportException specific to field caps? Does it need fixing long term?

Reply:
Sounds like something to dig deeper on and potentially fix as a follow-up, so that we can remove this line?

Reply:
Sorry if I'm being dense, but I'm not sure what the follow-up is. Do you mean to investigate why the transport layer returns different exceptions based on `skip_unavailable` status? Or do you mean not have special handling in field-caps to avoid logging this exception?

For the former, I suspect it is a bug in the transport layer handling of skip_unavailable=true cluster lookups. So we could file that. 

For the latter, even if the above issue is a bug and fixed, that means that we'd always get this ConnectTransportException whenever field-caps can't connect to a remote cluster. But field-caps always returns partial results (unless all field-caps lookups on all indexes and all clusters fail, which is rare), so even then I think  we'd want to have this code to avoid logging this 'error' since field-caps does best effort only?